src/peer_manager.rs

```diff
@@ -1200,6 +1202,25 @@ impl PeerManager {
         let _ = self.set_state(peer_id, PeerState::Routing(RoutingConnection::Direct));
     }
 
+    /// Get candidates of tunnel_node for us.
+    ///
+    /// To be qualified: peer must be directly conneted to us
+    ///                  it must be within NUM_TUNNEL_VIA_NODES closest to us
+    pub fn get_tunnel_candidates(&self) -> Vec<(XorName, PeerId)> {
```

Comment:
This function should get _all_ peers from our section before filtering, then after filtering, just take `NUM_TUNNEL_VIA_NODES` elements.